### PR TITLE
ci: aarch64-gnu-debug job

### PR DESCRIPTION
**Pull request by davidtwco**
- Adds a new CI job which checks that the compiler builds with `--enable-debug` and tests that `needs-force-clang-based-tests` pass (where cross-language LTO is tested).
- Add a test confirming that `-Zbranch-protection=pac-ret` and cross-language LTO work together.

r? @Mark-Simulacrum 

try-job: aarch64-gnu-debug

### CODE REVIEW

File: src/ci/docker/host-aarch64/aarch64-gnu-debug/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+# - That we can build the compiler with `--enable-debug`
+#   (without necessarily testing the result).
+# - That the tests with `//@ needs-force-clang-based-tests` pass, since they
+#   don't run by default unless RUSTBUILD_FORCE_CLANG_BASED_TESTS is set.
+#   - FIXME(https://github.com/rust-lang/rust/pull/126155#issuecomment-2156314273):
+#     Currently we only run the subset of tests with "clang" in their name.
+
+ENV SCRIPT \
+  python3 ../x.py --stage 2 build && \
+  python3 ../x.py --stage 2 test tests/run-make --test-args clang
```
Comment (jieyouxu) [inline src/ci/docker/host-aarch64/aarch64-gnu-debug/Dockerfile:57]: Remark: I need to document that this means that `//@ needs-force-clang-based-tests` is not alone sufficient, the test name must also contain a `clang` substring...

**Review by jieyouxu — APPROVED** [review]
The test itself LGTM

**Comment by Mark-Simulacrum** [comment]
@bors r+

This seems OK to me.

**Comment by tgross35** [comment]
Do we have a limit on the number of runners that handle arm jobs? The auto job is about an hour in and https://github.com/rust-lang-ci/rust/actions/runs/11309982802/job/31454499991 still says "Job is waiting for a runner from 'ubuntu-22.04-arm64-8core-32gb' to come online." for the existing aarch64-gnu job.

**Comment by tgross35** [comment]
Looks like this failed the rollup.

@bors r-

Regarding my above message - the original aarch64-gnu job did eventually start, just about an hour later than everything else.

**Comment by davidtwco** [comment]
@Mark-Simulacrum do you know whether we do have a limit on these runners or why it would be slow to start?

**Comment by Mark-Simulacrum** [comment]
There should be 50(?) runners worth of concurrency of this particular runner, so I doubt that was the problem. Looking back it looks like the job did eventually run, maybe this was fallout from the github problems last week (IIRC)?

I'd suggest we just retry.

**Comment by tgross35** [comment]
Did something land to fix the cross-lang-lto failure on this job from the rollup?

**Comment by jieyouxu** [comment]
FTR, the failure was

```
call.*rust_never_inlined
thread 'main' panicked at /checkout/tests/run-make/cross-lang-lto-clang/rmake.rs:41:10:
needle was not found in haystack
```

**Comment by jieyouxu** [comment]
That test was previously only ever exercised on the x86_64-debug counter-part, so presumably that is still going to fail.
@bors r-

**Comment by tgross35** [comment]
Just to have isolated logs

@bors try

File: src/ci/docker/host-aarch64/aarch64-gnu-debug/Dockerfile (hunk 1 above)
Comment (jieyouxu) [inline src/ci/docker/host-aarch64/aarch64-gnu-debug/Dockerfile:57]: Suggestion: could you leave me a FIXME to drop `--test-args clang` and fix the failing tests to make sure we don't have weird quirks like "only run-make tests with 'clang' in their name will be run by this job"?

Comment (davidtwco) [reply src/ci/docker/host-aarch64/aarch64-gnu-debug/Dockerfile:58]: Added

**Comment by davidtwco** [comment]
Oops, had a uncommitted chunk locally, should all pass now.

**Pull request merged**